common/app/assets/assets.scala

```diff
@@ -1,168 +1,135 @@
 package common.Assets
 
-import java.net.URL
 import common.{Logging, RelativePathEscaper}
 import conf.Configuration
 import org.apache.commons.io.IOUtils
 import play.api.libs.json._
 import play.api.{Mode, Play}
-import play.api.Play.current
 
 import scala.collection.concurrent.{Map => ConcurrentMap, TrieMap}
-import scala.util.control.NonFatal
+import scala.util.{Failure, Success, Try}
 
-case class Asset(path: String) {
-  val asModulePath = path.replace(".js", "")
-  lazy val md5Key = path.split('/').dropRight(1).last
+// turns an unhashed name into a name that's hashed if it needs to be
+class Assets(base: String, mapResource: String, useHashedBundles: Boolean = Configuration.assets.useHashedBundles) extends Logging {
 
-  override def toString = path
-}
-
-class AssetMap(base: String, assetMap: String) extends Logging {
+  lazy val lookup = Get(assetMap(mapResource))
```

Comment:
Can we be explicit about the type of `lookup`? It is not obvious when you read the code